src/dbstructureqitemviewfacade.cpp

```diff
@@ -0,0 +1,81 @@
+#include "DbStructureModel.h"
+#include "dbstructureqitemviewfacade.h"
+
+DbStructureQItemViewFacade::DbStructureQItemViewFacade(QAbstractItemView *aitemView) : QObject(aitemView)
+{
+    m_itemView = aitemView;
+}
+
+const QAbstractItemView *DbStructureQItemViewFacade::itemView() const
+{
+    return m_itemView;
+}
+
+bool DbStructureQItemViewFacade::hasSelection() const
+{
+    return itemView()->selectionModel()->hasSelection();
+}
+
+QString DbStructureQItemViewFacade::displayName()
+{
+    try {
```

Comment:
Why the exception handling here and in the functions below? I don't think any of the functions throw.